nucypher/blockchain/eth/actors.py

```diff
@@ -847,3 +846,146 @@ def request_evaluation(self, evidence):
+
+
+class MultiSigActor(NucypherTokenActor):
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.multisig_agent = ContractAgency.get_agent(MultiSigAgent, registry=self.registry)
+
+
+class Executive(MultiSigActor):
+    """
+    An actor having the power to authorize transaction executions to a delegated trustee.
+    """
+
+    class Authorization:
+        """
+        An edict issued by an Executive authorizing the execution of a multisig
+        transaction by the delegated trustee.
+        """
+
+        SignatureComponents = namedtuple('SignatureComponents', 'v r s')
+
+        def __init__(self, trustee_address: str, signed_transaction_hash: bytes):
+            self.trustee_address = trustee_address
+            self.__data = signed_transaction_hash
+            self.__r = None
+            self.__s = None
+            self.__v = None
+
+        def __bytes__(self):
+            pass
+
+        def get_signature_components(self) -> SignatureComponents:
+            components = self.SignatureComponents(v=self.__v, r=self.__r, s=self.__s)
+            return components
+
+        @property
+        def id(self):
+            # TODO - Unique ID
+            pass
+
+        def _serialize(self) -> bytes:
+            #TODO: BSS
+            pass
+
+        @classmethod
+        def _deserialize(cls, data: bytes) -> tuple:
+            #TODO: BSS
+            pass
+
+        def _write(self, filepath: str = None) -> str:
+            with open(filepath, 'wb') as file:
+                # TODO: Serialize
+                file.write(self.__data)
+            return filepath
+
+        @classmethod
+        def from_file(cls, filepath: str = None) -> 'Authorization':
+            with open(filepath, 'rb') as file:
+                data = file.read()
+                deserialized_data = cls._deserialize(data=data)
+                trustee_address, presigned_transaction_hash = deserialized_data
+            instance = cls(trustee_address=trustee_address,
+                           signed_transaction_hash=presigned_transaction_hash)
+            return instance
+
+
+    def sign_hash(self, unsigned_hash: bytes) -> bytes:
+        blockchain = self.multisig_agent.blockchain
+        signed_hash = blockchain.transacting_power.sign_hash(unsigned_hash=unsigned_hash)
+        return signed_hash
+
+    def authorize(self, trustee, contract_function: ContractFunction) -> Authorization:
+        try:
+            transaction = contract_function.buildTransaction()
+        except (TransactionFailed, ValueError):
+            # TODO - Handle Validation Failure
+            raise
+        unsigned_transaction_hash = self.multisig_agent.get_unsigned_transaction_hash(
+            trustee_address=trustee.checksum_address,
+            target_address=contract_function.address,
+            value=transaction['value'],
+            data=transaction['data'],
+            nonce=trustee.current_nonce
```

Comment:
Btw I think that with the nonce it should be resilient to chain reorgs

Reply:
Yeah - Its a good thought - What can we do to help ensure accuracy? 

Also - The `Trustee` can confirm consistency when combining `Authorization`s.